creators/Fenix/StopsCreatorFenix.py

```diff
@@ -1,19 +1,142 @@
 # coding=utf-8
 
+import overpy
+import sys
+
+from transitfeed import util
 from creators.StopsCreator import StopsCreator
+from osmhelper.osm_routes import Route, RouteMaster
+from osmhelper.osm_stops import Stop
 
 
 class StopsCreatorFenix(StopsCreator):
 
     def add_stops_to_schedule(self, schedule, data):
 
-        stops = data.stops
+        # Get stops information
+        stops = data.get_stops()
 
         # add all stops to GTFS
         for stop in stops.values():
+
+            # If there is no name, query one intelligently from OSM
+            if stop.name == Stop.NO_NAME:
+                self._get_interim_stop_name(stop)
+                print stop
+
+            # Add stop to GTFS object
             schedule.AddStop(
                 lat=float(stop.lat),
                 lng=float(stop.lon),
                 name=stop.name,
                 stop_id=str(stop.id)
             )
+
+        # Add loose stop objects to route objects
+        self.add_stops_to_routes(schedule, data)
+
+    def add_stops_to_routes(self, schedule, data):
+
+        routes = data.routes
+        stops = data.stops
+
+        # Loop through routes
+        for ref, route in routes.iteritems():
+            # Replace stop ids with Stop objects
+            self._fill_stops(stops, route)
+
+        data.routes = routes
+        return
+
+    def _fill_stops(self, stops, route):
+        """Fill a route object with stop objects for of linked stop ids
+
+        """
+        if isinstance(route, Route):
+            i = 0
+            for stop in route.stops:
+                if stop in stops:
+                    # Replace stop id with Stop objects
+                    route.stops[i] = stops[stop]
+                else:
+                    raise RuntimeError("Unknown stop: " + str(stop))
+                i += 1
+
+        elif isinstance(route, RouteMaster):
+            for route_variant_ref, route_variant in route.routes.iteritems():
+                self._fill_stops(stops, route_variant)
+
+        else:
+            raise RuntimeError("Unknown Route: " + str(route))
+
+    def _get_interim_stop_name(self, stop):
```

Comment:
Yes, those are the ones that make it (still) slow. But an optimization of this is clearly out of scope of this PR. This should be handled in a separate issue and though through very well. Then this could be also interesting not only for Fenix but also other networks.